src/Data/Text/IO.hs

```diff
@@ -176,17 +179,19 @@ hGetLine = hGetLineWith T.concat
 hPutStr :: Handle -> Text -> IO ()
 -- This function is lifted almost verbatim from GHC.IO.Handle.Text.
 hPutStr h t = do
-  (buffer_mode, nl) <-
-       wantWritableHandle "hPutStr" h $ \h_ -> do
-                     bmode <- getSpareBuffer h_
-                     return (bmode, haOutputNL h_)
+  (buffer_mode, nl, isUTF8) <-
+    wantWritableHandle "hPutStr" h $ \h_ -> do
+      bmode <- getSpareBuffer h_
+      let isUTF8 = maybe False ((== "UTF-8") . textEncodingName) $ haCodec h_
```

Comment:
When is `haCodec` not `Just`? I imagine if it's `Nothing`, it means that the file is open in binary mode, so it's fine to dump UTF-8 encoded data there as well, but I did not check.

Reply:
That seems like a breaking change. What would it have been before the change to utf8?